indexer-app/src/cmr/indexer/data/concepts/association_util.clj

```diff
@@ -2,19 +2,26 @@
   "Contains functions to parse and convert generic associations to a map structure 
   for each generic concept type."
   (:require
+   [clojure.set :as s]
    [cmr.common.concepts :as cc]))
 
 (defn allow-group-by-concept-type
   "Creates a temporary structure so that we can group the concept ids by concept type."
-  [concept-id]
-  {:concept-type (cc/concept-id->type concept-id)
-   :concept-id concept-id})
+  [association concept-id]
+  (let [asn (if (= concept-id (:source-concept-identifier association))
+              (-> association
+                  (dissoc :source-concept-identifier)
+                  (s/rename-keys {:associated-concept-id :concept-id}))
+              (-> association
+                  (dissoc :associated-concept-id)
+                  (s/rename-keys {:source-concept-identifier :concept-id})))]
+    (assoc asn :concept-type (cc/concept-id->type (:concept-id asn)))))
```

Comment:
I know you want to rename-keys to make it easier to convert it to the concept type. But I think you should restore the association in the end, i.e.  return (assoc association :concept-type (cc/concept-id->type (:concept-id asn))))), otherwise, you are not indexing the actual columns of the associations, which could make the code harder to maintain.  If you somehow need to use the concept-id field in the search result, it's better to convert it after getting the search result. More reason to add more integration tests for the new associations and association-details.